test/test_state_sparse_set.jl

```diff
@@ -171,6 +171,29 @@
         remove_all_but!(set, 2)
         @test length(set) == 1
         @test 2 ∈ set
+
+        set = HerbConstraints.StateSparseSet(HerbConstraints.StateManager(), BitVector((1, 0, 1)))
+        result = remove_all_but!(set, [1])
+        @test length(set) == 1
+        @test 1 ∈ set
+        @test result == true
+
+        set = HerbConstraints.StateSparseSet(HerbConstraints.StateManager(), BitVector((1, 0, 1)))
+        result = remove_all_but!(set, [2])
+        @test length(set) == 0
+        @test result == true
```

Comment:
@RichardWijers, can you double-check that this is the case? I think `result == false` here, right? Since `2` is not in the set.

Reply:
Partially addressed this by setting `result == false`, but I just realized this should probably `@test_throws` an `ArgumentException` because `2` is not a subset of `set`.